tests/test_cli_sync.py

```diff
@@ -128,6 +129,31 @@ def test_merge_error(req_lines, should_raise, runner):
         assert out.exit_code == 1
 
 
+@pytest.mark.parametrize(
+    "req_line",
+    (
+        "file:.",
+        "-e file:.",
+    ),
+)
+@mock.patch("piptools.sync.run")
+def test_merge_no_name_urls(run, req_line, runner):
+    """
+    Test sync succeeds when merging requirements that lack names.
+    """
+    reqs_paths = [
+        Path(name).absolute() for name in ("requirements.txt", "dev_requirements.txt")
```

Comment:
Would be nice to use `tmp_path` here to not touch the working directory

Reply:
Good idea. Maybe I'll make another pass through this module to update all the tests at once, unrelated to this change?

Reply:
Great, thanks 👍🏻